### ISSUE TITLE
services.AddMvc (Core 1.1) throwing exception after VS 15.8.0 update

### ISSUE THREAD

**Issue by EltonInAtlanta** [aspnet/Home#3435]
I have a project that is dotnet core 1.1 (and needs to stay there for the time being).  In my startup code I have the following:
```
        public void ConfigureServices(IServiceCollection services)
...
                services.AddMvc(config =>
                {
                    var policy = new AuthorizationPolicyBuilder().RequireAuthenticatedUser().Build();
                    config.Filters.Add(new AuthorizeFilter(policy));
                    config.Filters.Add(new CorsAuthorizationFilterFactory(Configuration["CorsPolicy"]));
                    config.Filters.Add(new RefreshJWTResultFilter(Options.Create(GetTokenOptions())));
                    config.Filters.Add(typeof(FaultTestingFilter));

                    config.Filters.Add(typeof(IncidentExceptionFilter));
                });
```
After updating Visual Studio to 15.8.0, this code throws the following exception:

```
"System.Collections.Generic.KeyNotFoundException"
The given key was not present in the dictionary.

```
With stacktrace:
```
   at System.Collections.Generic.Dictionary`2.get_Item(TKey key)
   at Microsoft.AspNetCore.Mvc.Internal.DefaultAssemblyPartDiscoveryProvider.CandidateResolver.ComputeClassification(String dependency)
   at Microsoft.AspNetCore.Mvc.Internal.DefaultAssemblyPartDiscoveryProvider.CandidateResolver.ComputeClassification(String dependency)
   at Microsoft.AspNetCore.Mvc.Internal.DefaultAssemblyPartDiscoveryProvider.CandidateResolver.ComputeClassification(String dependency)
   at Microsoft.AspNetCore.Mvc.Internal.DefaultAssemblyPartDiscoveryProvider.CandidateResolver.<GetCandidates>d__4.MoveNext()
   at System.Linq.Enumerable.<SelectManyIterator>d__17`2.MoveNext()
   at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
   at Microsoft.Extensions.DependencyInjection.MvcCoreServiceCollectionExtensions.GetApplicationPartManager(IServiceCollection services)
   at Microsoft.Extensions.DependencyInjection.MvcCoreServiceCollectionExtensions.AddMvcCore(IServiceCollection services)
   at Microsoft.Extensions.DependencyInjection.MvcServiceCollectionExtensions.AddMvc(IServiceCollection services)
   at Profile.Startup.ConfigureServices(IServiceCollection services)

```
I've simplified this to simply:
`services.AddMvc();`
and get the same error.

Source control indicates nothing in the startup file or package/reference list for this program has changed.
The project targets 4.6.1
`    <TargetFramework>net461</TargetFramework>
`
Here are the relevant packages:
```
    <PackageReference Include="Microsoft.AspNetCore.Authentication.JwtBearer" Version="1.1.2" />
    <PackageReference Include="Microsoft.AspNetCore.Authorization" Version="1.1.2" />
    <PackageReference Include="Microsoft.AspNetCore.Diagnostics" Version="1.1.2" />
    <PackageReference Include="Microsoft.AspNetCore.Identity" Version="1.1.2" />
    <PackageReference Include="Microsoft.AspNetCore.Mvc" Version="1.1.3" />
    <PackageReference Include="Microsoft.AspNetCore.Routing" Version="1.1.2" />
    <PackageReference Include="Microsoft.AspNetCore.Server.IISIntegration" Version="1.1.2" />
    <PackageReference Include="Microsoft.AspNetCore.Server.Kestrel" Version="1.1.2" />
    <PackageReference Include="Microsoft.DotNet.ProjectModel" Version="1.0.0-rc3-003121" />
    <PackageReference Include="Microsoft.Extensions.Configuration.EnvironmentVariables" Version="1.1.2" />
    <PackageReference Include="Microsoft.Extensions.Configuration.FileExtensions" Version="1.1.2" />
    <PackageReference Include="Microsoft.Extensions.Configuration.Json" Version="1.1.2" />
    <PackageReference Include="Microsoft.Extensions.Logging" Version="1.1.2" />
    <PackageReference Include="Microsoft.Extensions.Logging.Console" Version="1.1.2" />
    <PackageReference Include="Microsoft.Extensions.Logging.Debug" Version="1.1.2" />
    <PackageReference Include="Microsoft.Extensions.Logging.Filter" Version="1.1.2" />
    <PackageReference Include="Microsoft.Extensions.Options.ConfigurationExtensions" Version="1.1.2" />
    <PackageReference Include="Microsoft.IdentityModel.Tokens" Version="5.1.4" />

```
I'm very broken now, and really don't know how to proceed.  Any suggestions are very much appreciated.

**Comment by pranavkm** [comment]
cc @dsplaisted 

Seems like a dup of https://github.com/aspnet/Home/issues/3132, although that issue should have been fixed with the 2.1.400 SDK. Could you

a) Run `dotnet --info` from the command line and paste the contents
b) Build the app and paste the contents of `bin\Debug\net461\${YourAppName}.deps.json`  (a [gist](https://gist.github.com) works great for this)

In the meanwhile, https://github.com/aspnet/Home/issues/3132#issuecomment-388082725 has some possible workarounds to unblocks yourself.

**Comment by EltonInAtlanta** [comment]
I've attached the requested information.  I'm attempting to absorb the workarounds.  Thank you so much for responding.

[dotnet.info.txt](https://github.com/aspnet/Home/files/2294728/dotnet.info.txt)

[Sale.deps.json.txt](https://github.com/aspnet/Home/files/2294732/Sale.deps.json.txt)

**Comment by pranavkm** [comment]
@EltonInAtlanta I've reopened - https://github.com/aspnet/Mvc/issues/7983 - to see if we could get this patched. You could also consider short-circuiting MVC's application part discovery code that produces this error. Here's the code you would have to add to your Startup:

```C#
var manager = new ApplicationPartManager();
var assemblies = new[] { "Sale", "Horizon.Common.Tokens", "Mpp.Core.Infrastructure", "Mpp.Core.Sale" };

foreach (var name in assemblies)
{
    var assembly = Assembly.Load(new AssemblyName(name));
    manager.ApplicationParts.Add(new AssemblyPart(assembly));
}

services.AddSingleton(manager);
services.AddMvc();
```

This should avoid being pinned to an earlier version of the SDK.

**Comment by EltonInAtlanta** [comment]
@pranavkm I changed the file name on my global.json and dropped this code in to my startup.cs.  It works great.

**Comment by pranavkm** [comment]
Using https://github.com/aspnet/Mvc/issues/7983 to track further work.

**Comment by Gimgur** [comment]
Hi,

I'm new on github so sorry if it's the wrong way to do it.
I have exactly the same issue since .net Core SDK 2.1.400 and I was wondering how you define which assembly were causing it?

Thanks

**Comment by pranavkm** [comment]
@Gimgur, you'd want to add any assemblies that contain MVC controllers. If you need help with this, feel free to open a new issue in https://github.com/aspnet/Mvc.
